odoo/addons/base/models/res_company.py

```diff
@@ -376,24 +376,74 @@ def _get_main_company(self):
     def __accessible_branches(self):
         # Get branches of this company that the current user can use
         self.ensure_one()
-        children = self.search([('id', 'child_of', self.id)])
+
+        rslt = []
+        current = self.sudo()
         accessible = self.env.companies
```

Comment:
Just some style suggestion:
* `rslt` is a bit meh in general (doesn't really say what it is)
* `current` init would be better right before the loop since it is the variant of the loop

```suggestion
        accessible_ids = []
        all_accessible = self.env.companies
        current = self.sudo()
```